cppcache/src/TcrEndpoint.cpp

```diff
@@ -1155,6 +1157,10 @@ void TcrEndpoint::setConnectionStatus(bool status) {
       m_isActiveEndpoint = false;
       LOGFINE("Disconnected from endpoint %s", m_name.c_str());
       triggerRedundancyThread();
+      // Test hook
+      if (TcrEndpoint::TEST_DISCONNECTIONS) {
+        TcrEndpoint::listOfDisconnectedServers.push_back(m_name.c_str());
```

Comment:
If you are adding `std::string` to this vector then you should not be converting `m_name` to a `C-string` first.
```c++
TcrEndpoint::listOfDisconnectedServers.push_back(m_name);
```